lib/portlayer/attach/communication/connector.go

```diff
@@ -199,94 +261,124 @@ func (c *Connector) processIncoming(conn net.Conn) {
 		HostKeyCallback: callback,
 	}
 
-	log.Debugf("Initiating ssh handshake with new connection attempt")
-	var (
-		ccon    ssh.Conn
-		newchan <-chan ssh.NewChannel
-		request <-chan *ssh.Request
-	)
-
-	ccon, newchan, request, err = ssh.NewClientConn(conn, "", config)
+	// create the SSH connection
+	clientConn, chans, reqs, err := ssh.NewClientConn(conn, "", config)
 	if err != nil {
 		log.Errorf("SSH connection could not be established: %s", errors.ErrorStack(err))
 		return
 	}
 
-	client := ssh.NewClient(ccon, newchan, request)
-
-	var ids []string
-	ids, err = SSHls(client)
+	// ask the IDs
+	ids, err := ContainerIDs(clientConn)
 	if err != nil {
 		log.Errorf("SSH connection could not be established: %s", errors.ErrorStack(err))
 		return
 	}
 
-	var si SessionInteraction
+	// Handle global requests
+	go c.reqs(reqs, clientConn, ids)
+	// Handle channel open messages
+	go c.chans(chans)
+
+	// create the connections
+	c.ids(clientConn, ids)
```

Comment:
ah - this may be why the runblocking on all sessions isn't impact us currently. **I'm unsure what happens if we don't drain the stdout streams of the "un-attached" connections in this model**

We should think about, then document, what the behaviour _should_ be in the case of multiple sessions where we only care about output from one of them.